### DISCUSSION TITLE
Hardware Acceleration?

### DISCUSSION THREAD

**Discussion by Injuns4Trump** [GitSquared/edex-ui#1129] [Questions]
Is there any way to enable hardware/GPU acceleration in eDEX-UI so that it takes up less CPU usage? At first I thought this was a problem with Linux but it doesn't seem to use acceleration on the Windows version either.

Thanks

**Comment by chromer030** [comment]
edex-ui is an Electron app and Electron apps automatically use HW acceleration whenever & wherever possible.

**Comment by GitSquared** [comment]
> @chromer030: edex-ui is an Electron app and Electron apps automatically use HW acceleration whenever & wherever possible.

This is mostly true but the app's code itself has a lot of impact on that "wherever possible".

On Windows, HW acceleration is well supported and since we use WebGL/Canvas drawing in the UI on the globe & realtime graphs, it provides a nice boost.
However, fetching all the system information that is displayed is complicated, and we need to spawn WMIC processes, which can cause lag & thread starving - that we can only mitigate so much before the UI becomes more of a PowerPoint deck rather than a live monitoring tool.

On Linux, reading system information is easy & fast via the `/proc` virtual file system, however 3D HW acceleration is not well supported.
Chromium, which is responsible for managing rendering in the Electron stack, maintains a "blocklist" of GPUs that have not been tested by their team to work well with hardware acceleration. We can bypass this list and force-enable HW acceleration using flags embedded in the app's source code - eDEX actually does not use them yet, I'm going to add some.

Edit: added blocklist bypass flags in 9cfc2e2. You can grab builds [here](https://github.com/GitSquared/edex-ui/actions/runs/898828242#artifacts), would love to know if it helps with perf on linux for you.
